include/openthread/history_tracker.h

```diff
@@ -124,6 +126,38 @@ typedef struct otHistoryTrackerMessageInfo
     bool       mRadioTrelUdp6 : 1;   ///< Indicates whether msg was sent/received over a TREL radio link.
 } otHistoryTrackerMessageInfo;
 
+/**
+ * This enumeration defines the events in a neighbor info (i.e. whether neighbor is added, removed, or changed).
+ *
+ * Event `OT_HISTORY_TRACKER_NEIGHBOR_EVENT_RESTORING` is applicable to child neighbors only. It is triggered after
+ * the device (re)start when the previous children list is retrieved from non-volatile settings and decide tries to
+ * restore connection to them.
+ *
+ */
+typedef enum
+{
+    OT_HISTORY_TRACKER_NEIGHBOR_EVENT_ADDED     = 0, ///< Neighbor is added.
+    OT_HISTORY_TRACKER_NEIGHBOR_EVENT_REMOVED   = 1, ///< Neighbor is removed.
+    OT_HISTORY_TRACKER_NEIGHBOR_EVENT_CHANGED   = 2, ///< Neighbor changed (e.g., device mode flags changed).
+    OT_HISTORY_TRACKER_NEIGHBOR_EVENT_RESTORING = 3, ///< Neighbor is being restored (applicable to child only).
+} otHistoryTrackerNeighborEvent;
+
+/**
+ * This structure represents a neighbor info.
+ *
+ */
+typedef struct otHistoryTrackerNeighborInfo
+{
+    otExtAddress mExtAddress;           ///< Neighbor's Extended Address.
+    uint16_t     mRloc16;               ///< Neighbor's RLOC16
+    int8_t       mAverageRssi;          ///< Average RSSI of rx frames from neighbor at the time of recording entry.
+    uint8_t      mEvent : 2;            ///< Indicates the event (`OT_HISTORY_TRACKER_NEIGHBOR_EVENT_*` enumeration).
```

Comment:
nit: be consistent with `.` at end of line.

Reply:
Thanks for checking and noticing these 👍